server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -308,6 +312,15 @@ protected void doClose() throws IOException {
         indicesRefCount.decRef();
     }
 
+    /**
+     * Wait for this {@link IndicesService} to be effectively closed. When this returns, all shard and shard stores are closed and all
```

Comment:
I think the comment is not entirely accurate? The guarantees described here only hold if this method returns `true`. If it returns `false` it timed out waiting for the condition to happen.